pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -427,8 +427,16 @@ func (f *framework) ApplyScoreWeights(pc *PluginContext, pod *v1.Pod, scores Plu
 			errCh.SendErrorWithCancel(err, cancel)
 			return
 		}
-		for i := range nodeScoreList {
-			nodeScoreList[i] = nodeScoreList[i] * weight
+
+		for i, score := range nodeScoreList {
+			// Skip score plugin if it returns invalid score.
+			if score > NodeScoreMax || score < NodeScoreMin {
+				klog.Errorf("Skip score plugin %q since it returns an invalid score %q, it should in the range of [0, 100] after normalizing.", pl.Name(), score)
+				delete(scores, pl.Name())
+				return
```

Comment:
I think we should also send an error over the `errCh`. We currently ignore a faulty plugin and only log a message. These log messages may remain unnoticed for a while. I would prefer a more serious action, like not scheduling the pod so that a cluster admin notices the issue.
If you decided to go with my suggestion, you may want to remove the `klog.Errorf` above and only send an error message over `errCh`.

Reply:
SG